src/tdd/isograma.ts

```diff
@@ -1,9 +1,24 @@
-function isograma(string) {
+   function isograma(s: string): boolean {
 
-        if (string !== '-') {
-          callback(string);
-        }}
+        
+   
+        let letraComEsp = s.replaceAll("-", "");
+            letraComEsp = letraComEsp.replaceAll(" ", "");
+        // remove caracteres "-" e ""
 
+        let vistos: Array<string> = [];
+        for (let i = 0; i < s.length; i++) {
+        vistos.push(s[i]);
+        // objeto vistos esta armazenando cada informação da string do teste
+        // for loop 
+    
+        if (vistos.length > 1)
+            return false
+            //se a primeira letra existe na string declarada
+        else (vistos.length < -1);
```

Comment:
Usando a variável `vistos` como um array dessa forma não vai dar muito certo essa lógica.
O `for`sempre roda 1 vez e retorna true. Lembra de rodar o comando `yarn prettier` e usar `{...}` para mostrar o que está "dentro" dos seus `if`s, `else`s, `for`s e tal, isso facilita muito a leitura do código.

Para a variável `vistos` eu recomendo usar um Map:
```js
type Vistos = {
    [key: string]: boolean;
};

let vistos: Vistos = {};
```

E lembra de checar pela string vazia, e tornar todas as letras da string maiúsculas ou minúsculas.